apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -159,6 +176,12 @@ where
     // TODO(@TylerBloom): Add in once defer is supported.
     #[serde(skip)]
     defer_on_tail: Option<DeferDirectiveArguments>,
+    /// At the point where a `@context` is set, we will have fields to select
+    context_to_selection: Vec<Option<IndexSet<String>>>,
```

Comment:
I changed this to a HashSet. Is that what you had in mind? I really only need the name of the `entry.id` part in this map.

Reply:
No, I mean that all of those field are vectors of options of some data. I'm asking if it would make sense to merge them together and have one field that is a single vector of tuples (or some new type).